eland/score.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2020 Elasticsearch BV
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class RandomScore:
```

Comment:
I think this can actually be a subclass of `QueryFilter` instead of it's own thing. Definitely see the issue with the "what happens if I combine a `RandomScoreFilter` with no query, we should add a `MatchAllFilter(QueryFilter)` that basically is `{"match_all": {}}` and use that if we detect `self._query.empty()` within `Query.random_score()`.

All of this definitely makes me think that "Filter" is the wrong word to use here since not only are we filtering things we also augmenting queries. Will think about this more. :)